uni/udb/breakpoint.icn

```diff
@@ -76,75 +75,107 @@ end
 
    if type(cmd) == "list" then name := cmd[3]
    msg := ""
    if /name then {
       every x := key(breakPoints) do {
-	 L := breakPoints[x]
-	 every bp := !L do {
-            msg ||:= constructBpMsg(bp)
-            }
-	 }
-      }
+         L := breakPoints[x]
+         every bp := !L do {
+                  bpTable := constructBpMsg(bp)
+                  msg ||:= bpTable["consoleMsg"]
+                  push(resultTable["breakpoints"], bpTable)
+         }
+	   }
+   }
    else {
       if integer(name) then {
-	 every x := key(breakPoints) do {
-	    L := breakPoints[x]
-	    every bp := !L do {
-	       if name = bp.id then {
-                  msg := constructBpMsg(bp)
-	          break break
-	          }
-	       }
-	    msg||:="\n   Breakpoint #"||name||" is not available."
-	    }
-	 }
+	      every x := key(breakPoints) do {
+            L := breakPoints[x]
+            every bp := !L do {
+               if name = bp.id then {
+                  bpTable := constructBpMsg(bp)
+                  msg ||:= bpTable["consoleMsg"]
+                  push(resultTable["breakpoints"], bpTable)
+                  break break
+               }
+            }
+            msg||:="\n   Breakpoint #"||name||" is not available."
+            push(resultTable["breakpoints"], bpTable)
+	      }
+	   }
       else {
-	 if not find(".icn",name) then name ||:= ".icn"
-	 if DState.srcFile.isSrcFile(name) then {
-	    if L:=breakPoints[name] & type(L) == "list" then {
-	       every bp := !L do
-                  msg ||:= constructBpMsg(bp)
-	       }
-	    else
-	       msg := "\n   No breakpoints available in "||name
-	    }
-	 else
-	    msg := "\n   Source file \""||name||"\" is not in the binary !!!"
-	 }
-      }
-   if *msg = 0 then
+         if not find(".icn",name) then name ||:= ".icn"
+         if DState.srcFile.isSrcFile(name) then {
+            if L:=breakPoints[name] & type(L) == "list" then {
+               every bp := !L do {
+                  bpTable := constructBpMsg(bp)
+                  msg ||:= bpTable["consoleMsg"]
+                  push(resultTable["breakpoints"], bpTable)
+               }
+            }
+	         else {
+               msg := "\n   No breakpoints available in "||name
+               resultTable["success"] := "__true__"
+               resultTable["message"] := "No breakpoints available in " ||name
+            }
+         }
+         else {
+            msg := "\n   Source file \""||name||"\" is not in the binary !!!"
+            resultTable["success"] := "__false__"
+            resultTable["message"] := "Source file " ||name|| " not in binary"
+            resultTable["reason"] := "failed"
+         }
+	   }
+   }
+   if *msg = 0 then {
       msg := "\n   No breakpoints available."
-   DState.Write(msg)
+      resultTable["consoleMsg"] := msg
+      resultTable["success"] := "__true__"
+      resultTable["message"] := "No breakpoints available"
+   }
+   resultTable["consoleMsg"] := msg
```

Comment:
doesn't this overwrite the value set in line 165 above?.

Also, since setting these table fields is a common pattern, would it be better to have that as a function, i.e:
```
setResultTable(resultTable,  msg,  "__true__",  "No breakpoints available")  
```

Reply:
Does Unicon have a keyword parameter passing mechanism? It would be useful in implementing something like this.

Reply:
Implemented